typescript.md

```diff
@@ -204,10 +223,60 @@ export interface User { ... }
 ```ts
 interface Building {
   room: {
-    door: string,
-    walls: string[],
+    door: string;
+    walls: string[];
   };
 }
 
 type Walls = Building['room']['walls']; // string[]
 ```
+
+## Keyof Type Operator
+
+```ts
+type Point = { x: number; y: number };
+
+type P = keyof Point; // x | y
+```
+
+## Conditinal Types
+
+```ts
+// SomeType extends OtherType ? TrueType : FalseType;
+
+type ToArray<T> = T extends any ? T[] : never;
+
+type StrArrOrNumArr = ToArray<string | number>; // string[] | number[]
+```
+
+### Inferring
+
+```ts
+type GetReturnType<T> = T extends (...args: unknown[]) => infer R
+  ? R
+  : never;
+
+type Num = GetReturnType<() => number>; // number
+
+type First<T extends Array<any>> = T extends [infer F, ...infer Rest] ? F : never;
+
+type Str = First<['hello', 1, false]>; // 'hello'
+```
+
+## Literal Type
+
+```ts
+const point = { x: 4, y: 2 }; // { x: number, y: number }
+
+const literalPoint = point as const; // { x: 4, y: 2 };
```

Comment:
Hello,

I've the error :

```
const literalPoint = point as const;
                     -----
A 'const' assertions can only be applied to references to enum members, or string, number, boolean, array, or object literals.ts(1355)
```

Reply:
Hello, @lmarvaud 

I fixed it, It was a incorrect example of the a 'const' assertion